pkg/ccl/backupccl/restore_job.go

```diff
@@ -2862,12 +2872,16 @@ func (r *restoreResumer) restoreSystemUsers(
 			return err
 		}
 
-		insertRoleMember := `INSERT INTO system.role_members ("role", "member", "isAdmin") VALUES ($1, $2, $3)`
+		insertRoleMember := `INSERT INTO system.role_members ("role", "member", "isAdmin", role_id, member_id) VALUES ($1, $2, $3, $4, $5)`
 		for _, roleMember := range roleMembers {
+			role := tree.MustBeDString(roleMember[0])
+			member := tree.MustBeDString(roleMember[1])
+			isAdmin := tree.MustBeDBool(roleMember[2])
 			// Only grant roles to users that don't currently exist, i.e., new users we just added
-			if _, ok := newUsernames[roleMember[1].String()]; ok {
-				if _, err = executor.Exec(ctx, "insert-non-existent-role-members", txn, insertRoleMember,
-					roleMember[0], roleMember[1], roleMember[2]); err != nil {
+			if newUsernames[member.String()] {
+				if _, err := executor.Exec(ctx, "insert-non-existent-role-members", txn, insertRoleMember,
+					role, member, isAdmin, userIDMap[role], userIDMap[member],
```

Comment:
I wonder if we could do the ID lookup directly in SQL so that we don't need to keep this userIDMap in memory.  This function already does a bunch of bufferred queries, so I'm fine with the map if you think it ends up cleaner this way.

Reply:
Changed